sky/backends/backend_utils.py

```diff
@@ -1809,7 +1809,8 @@ def _update_cluster_status_no_lock(
 
 def _update_cluster_status(
```

Comment:
I do not clearly remember the differences between `_update_cluster_status()` and `refresh_cluster_status_handle()`. After reading the docstrings, still having some troubles telling the differences.

The `status -r` code path calls the former:

https://github.com/skypilot-org/skypilot/blob/c86b69dc3813d8923824ceb0e9f5840c11309339/sky/backends/backend_utils.py#L2046-L2054

Would this change be needed if it calls the latter?  Which already had identity checks.

Reply:
The main difference is the former returns the entire record but the latter returns the (status, handle) tuple. If we let the refresh call the later, we may have to add a argument to the latter function to return the record. I would prefer the current implementation, since the `refresh_cluster_status_handle` is public and used a lot in other modules.

Reply:
I refactored it a bit, by adding a new function `refresh_cluster_record`, and all the callers will go through this function now. Does that look better @concretevitamin? : )